### PR TITLE
Document how to install Pillow from a directory

### PR DESCRIPTION
**Pull request by radarhere**
Suggestion for #6755

### CODE REVIEW

**Comment by hugovk** [comment]
The page goes a bit like this:

```rst
Building From Source
--------------------

External Libraries
^^^^^^^^^^^^^^^^^^

[General info about dependencies]

[Then straight onto how to install from source from PyPI, archive, or Git]
```
```rst
Build Options
^^^^^^^^^^^^^

[Info about options]
```
```rst
Building on macOS
^^^^^^^^^^^^^^^^^

[How to install dependencies]

[How to install from source from PyPI, or archive]
```
```rst
Building on Windows
^^^^^^^^^^^^^^^^^^^

[Use wheels, or check `winbuild`]
```
```rst
Building on Windows using MSYS2/MinGW
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[How to install dependencies]

[How to install from source from PyPI]
```
```rst
Building on FreeBSD
^^^^^^^^^^^^^^^^^^^

[How to install dependencies]

[How to install from source from PyPI]
```
```rst
Building on Linux
^^^^^^^^^^^^^^^^^

[How to install dependencies]
```
```rst
Building on Android
^^^^^^^^^^^^^^^^^^^

[How to install dependencies]
```

So we have the install instructions repeated in the External Libraries and in some of the Building sections. 

Can we tidy this up a bit somehow?

**Comment by radarhere** [comment]
Ok, see what you think of this idea - I've moved the build instructions for Pillow itself after all of the external library instructions.

File: docs/installation.rst [hunk 1]
```diff
@@ -406,6 +323,84 @@ environment. The dependencies can be installed by::
...
+If you would like to install from a local copy of the source code instead, you
+can download and extract the `compressed archive from PyPI`_, or clone from
+GitHub with ``git clone https://github.com/python-pillow/Pillow``.
+
+After navigating to the Pillow directory, run::
+
+    python3 -m pip install --upgrade pip
+    python3 -m pip install .
+
+.. _compressed archive from PyPI: https://pypi.org/project/Pillow/
```
Comment (hugovk) [inline docs/installation.rst:359]: And link directly to the files page:
```suggestion
.. _compressed archive from PyPI: https://pypi.org/project/Pillow/#files
```

**Review by hugovk — COMMENTED** [review]
Looks better, thanks!

**Pull request merged**